packages/js/settings-editor/src/route.tsx

```diff
@@ -38,20 +38,19 @@ const NotFound = () => {
 /**
  * Default route when active page is not found.
  *
- * @param {string}       activePage   - The active page.
- * @param {settingsData} settingsData - The settings data.
- *
+ * @param {string}        activePage - The active page.
+ * @param {SettingsPages} pages      - The settings pages.
```

Comment:
@chihsuan The linter is giving me this error even though `SettingsPages` is defined in `global.d.ts`. Do you know what is required to get eslint to respect globals?

```
42:1  error  The type 'SettingsPages' is undefined  jsdoc/no-undefined-types`
```

Reply:
Ah, it looks like ESL int's JSDoc checker doesn't automatically recognize types defined in global.d.ts. We have to use import syntax or configure ESLint to recognize global types like:

```js
module.exports = {
  rules: {
    'jsdoc/no-undefined-types': ['error', {
      definedTypes: [
        // Add your global types here
        'SettingsPages',
        'SettingsPage',
        'SettingsData'
      ]
    }]
  }
}
```

I see that the other JSdoc params don't have the same issue because we use `settingsPage` instead of `SettingsPage`. I think JSDoc checker thinks `settingsPage` is a variable so doesn't throw any error but it's actually be considered as `any` type.

Reply:
I think we can move the types from global.d.ts to the typings instead. It will need to import the types in the files that need them, but it's more explicit and can be imported if someone needs them when using the package. What do you think? I can make the change.

Reply:
Oh wow, ok yeah thats not ideal. Thank you! I lost a good chunk of time with this one already.

>I think we can move the types from global.d.ts to the typings instead.

I'm trying to see if `typings/global.d.ts` can be added in the configuration somehow, but no luck so far.

If that doesn't work, I think its ok to be explicit and move the types.